mlir/include/mlir/Dialect/Vector/TransformOps/VectorTransformOps.td

```diff
@@ -68,6 +68,22 @@ def ApplyRankReducingSubviewPatternsOp : Op<Transform_Dialect,
   let assemblyFormat = "attr-dict";
 }
 
+def ApplyDropUnitDimWithShapeCastPatternsOp : Op<Transform_Dialect,
+    "apply_patterns.vector.drop_unit_dims_with_shape_cast",
+    [DeclareOpInterfaceMethods<PatternDescriptorOpInterface>]> {
+  let description = [{
+     Apply vector patterns to fold unit dims with vector.shape_cast Ops:
+      - DropUnitDimFromElementwiseOps
+      - DropUnitDimsFromScfForOp
+      - DropUnitDimsFromTransposeOp
```

Comment:
I'm not a big fan of adding the patterns to the documentation because it could be outdated quickly. It is fine to me because other operations are following the convention...

Reply:
I also struggle with this a bit. 

I do like that it helps to find duplicates. Also, makes the overall description a bit easier and more concise (provided that these patterns have good descriptive names).